src/AppInstallerCommonCore/Public/winget/UserSettings.h

```diff
@@ -130,6 +136,10 @@ namespace AppInstaller::Settings
         SETTINGMAPPING_SPECIALIZATION(Setting::NetworkDOProgressTimeoutInSeconds, uint32_t, std::chrono::seconds, 60s, ".network.doProgressTimeoutInSeconds"sv);
         SETTINGMAPPING_SPECIALIZATION(Setting::InstallLocalePreference, std::vector<std::string>, std::vector<std::string>, {}, ".installBehavior.preferences.locale"sv);
         SETTINGMAPPING_SPECIALIZATION(Setting::InstallLocaleRequirement, std::vector<std::string>, std::vector<std::string>, {}, ".installBehavior.requirements.locale"sv);
+        SETTINGMAPPING_SPECIALIZATION(Setting::InstallIgnoreWarnings, bool, bool, false, ".installBehavior.ignoreWarnings"sv);
+        SETTINGMAPPING_SPECIALIZATION(Setting::PortableAppUserRoot, std::string, std::string, {}, ".installBehavior.portableAppUserRoot"sv);

```

Comment:
I think that these settings should return a `std::filesystem::path` to prevent UTF-8 to path conversion bugs.  The path should also be verified to be good.  That should allow the settings fallback mechanism to kick in if the path is not proper; it needs to be a full path, not relative, etc.

Also, if you don't manually convert to the UTF-8 string to UTF-16, you will get the above bug (which you have in `Runtime.cpp`).

Reply:
changed settings to return a path as well as added settings validation to ensure that the path is absolute. Also added a test case checking this behavior when providing a relative path.